### PR TITLE
Fix telegram bot documentation

### PR DESCRIPTION
**Pull request by vigonotion**
## Proposed change

"Telegram chatbot" should be called "Telegram bot", like the corresponding integration. The link to source was broken because of this, that's why I had to change the filename as well.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase:
- This PR fixes or closes issue:

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

**Review by klaasnicolaas — CHANGES_REQUESTED** [review]
We should also change this line then: https://github.com/home-assistant/home-assistant.io/blob/next/source/_redirects#L1888

File: source/_redirects [hunk 1]
```diff
@@ -1887,7 +1887,7 @@
 /components/ted5000 /integrations/ted5000
 /components/teksavvy /integrations/teksavvy
 /components/telegram /integrations/telegram
-/components/telegram_bot /integrations/telegram_chatbot
+/components/telegram_bot /integrations/telegram_bot
```
Comment (klaasnicolaas) [inline source/_redirects:1890]:
```
/integrations/telegram_chatbot /integrations/telegram_bot
```

Maybe add also this rule

**Review by klaasnicolaas — CHANGES_REQUESTED** [review]

**Review by frenck — CHANGES_REQUESTED** [review]
Please update all internal links/references on our website as well.
Redirects are fine for handling external links, but our internal links should be correct.

**Review by frenck — APPROVED** [review]
LGTM! ✅  thanks!

**Pull request merged**